code-challenges.test.js

```diff
@@ -68,6 +109,26 @@ const accountTransactions2 = [250, -89, 100, -96]
 const accountTransactions3 = []
 // Expected output: []
 
+describe("accumulate", () => {
+  it("takes in an array and returns a new array of the accumulating sum", () => {
+    expect(accumulate(6=accountTransactions1)).toEqual([100, 83, 60, 51])
+    expect(accumulate(accountTransactions2)).toEqual([250, -89, 100, -96])
+    expect(accumulate(accountTransactions3)).toEqual([])
+  })
+})
+
 // b) Create the function that makes the test pass.
 
 // Pseudo code:
+// function name: accumulate
+// input: array
+// output: array of accumulating sum
+// process: .reduce() method
+
+// I got stuck on this problem because after doing some research, I am pretty sure I am supposed to use the .reduce() method, but I'm having trouble understanding the concept of the callback function, and how to use the built-in values. I don't understand if the callback function is a seperate function that I write and then plug into the parentheses after the reduce method, or is it something that is just built in and can be used by just putting values in the parantheses? 
+
+const accumulate = (array) => {
+  array.reduce()
+  // I think this is where I would use the callback function but I don't understand the syntax.
```

Comment:
If we look at .map, we can see a callback function there: `array.map((value, index) => {return value})`
The `(value, index) => {return value}` part is a callback function.
`.reduce` also takes a callback function, just with different parameters.
`const accumulate = (array) => { let sum = 0; return array.reduce((accumulator, currentValue) => { return accumulator = [...accumulator, sum += currentValue] }, []) }`

That is one approach you could use if you wanted to use `.reduce`. The callback function takes the parameters of an accumulator, which gets updated each iteration, and the currentValue, which is the current value in the iteration. It then returns what you want to update the accumulator with, in this case, an array with everything already in the accumulator, and adding in the sum of the `sum` variable and the current value. The `, []` part defines what you want the accumulator to start off as.

This is a bit complex, so if you want a deeper explanation definitely bring it up during assessment review or reach out to me on Slack on what you're confused on 🙂